native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -16,139 +16,111 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-
-import {
-  StyleSheet,
-  TouchableWithoutFeedback,
-  View,
-  ViewPropTypes,
-} from 'react-native';
+/* @flow */
 
+import React from 'react';
 import {
-  borderRadiusSm,
-  colorBlue500,
+  spacingMd,
+  spacingBase,
+  spacingLg,
+  colorRed500,
   colorGray50,
   colorGray500,
   colorGreen500,
-  colorRed500,
   colorYellow500,
-  spacingBase,
-  spacingLg,
-  spacingSm,
-  spacingXxl,
+  borderRadiusSm,
 } from 'bpk-tokens/tokens/base.react.native';
-
-import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
-import BpkIcon, { icons } from 'react-native-bpk-component-icon';
 import BpkText from 'react-native-bpk-component-text';
+import { StyleSheet, View } from 'react-native';
+import BpkIcon, { icons } from 'react-native-bpk-component-icon';
+import BpkButtonLink from 'react-native-bpk-component-button-link';
+import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
 import BpkTouchableNativeFeedback from 'react-native-bpk-component-touchable-native-feedback';
 
-import AnimateAndFade from './AnimateAndFade';
-import ALERT_TYPES from './AlertTypes';
 import {
-  dismissablePropType,
-  toggleExpandedButtonLabelPropType,
-  dismissableLabelPropType,
-} from './customPropTypes';
+  type Props,
+  ALERT_TYPES,
+  commonPropTypes,
+  commonDefaultProps,
+} from './common-types';
 
-const styles = StyleSheet.create({
-  mainContainer: {
-    paddingVertical: spacingBase,
-    paddingHorizontal: spacingLg,
+import AnimateAndFade from './AnimateAndFade';
+
+const iconAlignmentPadding = {
+  // Current android "sm" text calculated line height is 19.4.
```

Comment:
As mentioned in this PR description - the android icon alignment with text size small is just plain off. That's why i have kept some hard coded magic numbers.

Reply:
What kind of alignment dose this achieve? Center? Why doens't `alignItems: 'center'` work in that case?

Reply:
`alignItems: 'center' work in that case?` does work (sort of - the alignment isn't perfect). But a problem arises when the banner alert message spans multiple lines. Here is a graphic which shows the alignment and what happens with a wrapping message:

![image](https://user-images.githubusercontent.com/1440090/38204338-7e7ebda8-369a-11e8-86c2-d12ad02b918c.png)